src/main/resources/public/ts/sniplets/content/workspace-nextcloud-content.sniplet.ts

```diff
@@ -142,33 +143,33 @@ class ViewModel implements IViewModel {
         let folderContent: any = angular.element(element).scope();
         if (folderContent && folderContent.folder) {
             if (folderContent.folder instanceof models.Element) {
-                this.moveDocumentToWorkspace(folderContent.folder, document)
-                    .then(async (_: AxiosResponse) => {
-                        return nextcloudService.listDocument(model.me.userId, selectedFolderFromNextcloudTree.path ?
-                            selectedFolderFromNextcloudTree.path : null);
-                    })
-                    .then((syncedDocument: Array<SyncDocument>) => {
-                        this.documents = syncedDocument
-                            .filter((syncDocument: SyncDocument) => syncDocument.path != selectedFolderFromNextcloudTree.path)
-                            .filter((syncDocument: SyncDocument) => syncDocument.name != model.me.userId);
-                        this.safeApply();
-                    })
-                    .catch((err: AxiosError) => {
-                        const message: string = "Error while attempting to move nextcloud document to workspace " +
-                            "or update nextcloud list";
-                        console.error(message + err.message);
-                    });
+                let fileToMove : Set<SyncDocument> = new Set(this.selectedDocuments).add(document);
+                let finalUpload : Array<String> = Array.from(fileToMove).filter(file => !file.isFolder).map(file => file.path);
+                if (finalUpload.length > 0) {
```

Comment:
finalUpload.length

Reply:
et vérifie bien que quoi qu'il arrive, finalUpload sera toujours un tableau sinon tu risques un 'cannot read property of undefined'

Reply:
Le tableau sera toujours défini même si le set est nul donc aucun risque de 'cannot read property of undefined'